### PR TITLE
`add_new_user`, `update_user_balance`, and `update_user_role`

### PR DESCRIPTION
**Pull request by vtomole**
Follow up for https://github.com/SupertechLabs/SuperstaQ/pull/1870

Example

```python
import qiskit_superstaq as qss

superstaq = qss.superstaq_provider.SuperstaQProvider(remote_host="https://127.0.0.1:5000")

print(superstaq.add_new_user("victory", "vtomole@gmail.com"))
# prints
# The user has been added with token
print(superstaq.update_user_balance("vtomole@gmail.com", 5.00))
# prints
# The account's balance has been updated
print(superstaq.update_user_role("vtomole@gmail.com", 5))
# prints
# The account's role has been updated
```

### CODE REVIEW

**Review by paaige — COMMENTED** [review]
I might be missing context but making these requests requires a higher level of authentication right? Previously they weren't exposed in the API

**Comment by vtomole** [comment]
@paaige Yep, only members of Super.tech can use these endpoints.

File: general_superstaq/superstaq_client.py [hunk 1]
```diff
@@ -196,6 +196,39 @@ def ibmq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
...
         return self.post_request("/ibmq_token", json_dict)
 
+    def add_new_user(self, json_dict: Dict[str, str]) -> str:
+        """Makes a POST request to SuperstaQ API to add a new user in database.
+
+        Args:
+            json_dict: dictionary with user entry.
+
+        Returns:
+            The json body of the response as a dict.
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/superstaq_client.py]: Done.

File: general_superstaq/superstaq_client.py [hunk 2]
```diff
@@ -196,6 +196,39 @@ def ibmq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
 
         return self.post_request("/ibmq_token", json_dict)
 
+    def add_new_user(self, json_dict: Dict[str, str]) -> str:
+        """Makes a POST request to SuperstaQ API to add a new user in database.
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/superstaq_client.py]: Done.

File: general_superstaq/superstaq_client.py [hunk 3]
```diff
@@ -196,6 +196,39 @@ def ibmq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
...
+        Args:
+            json_dict: dictionary with user entry and new balance.
+
+        Returns:
+            The json body of the response as a dict.
+        """
+        return self.post_request("/update_user_balance", json_dict)
+
+    def update_user_role(self, json_dict: Dict[str, Union[int, str]]) -> str:
+        """Makes a POST request to SuperstaQ API to update a user's role in database.
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/superstaq_client.py]: Done.

**Comment by vtomole** [comment]
@PalashGoiporia Is there a way for us to ignore adding docstrings for tests? Cirq doesn't have it.

**Comment by pgoiporia** [comment]
> @PalashGoiporia Is there a way for the linter to to ignore adding docstrings for tests without the label? Cirq doesn't require documenting tests.

I just drop an ignore at the top of the file and it takes care of it.

File: general_superstaq/superstaq_client.py [hunk 4]
```diff
@@ -191,11 +191,44 @@ def ibmq_set_token(  # pylint: disable=missing-param-doc,differing-param-doc
             ibmq_token: dictionary with IBMQ token string entry.
 
         Returns:
-            The json body of the response as a dict.
+            The response as a string.
         """
 
         return self.post_request("/ibmq_token", json_dict)
 
+    def add_new_user(self, json_dict: Dict[str, str]) -> str:
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/superstaq_client.py:185]: Done.

File: general_superstaq/user_config_test.py [hunk 5]
```diff
@@ -32,9 +32,52 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
...
     )
     service = gss.user_config.UserConfig(client)
-    assert service.ibmq_set_token("valid token") == {
-        "status": "Your IBMQ account token has been updated"
-    }
+    assert service.ibmq_set_token("valid token") == "Your IBMQ account token has been updated"
+
+
+@mock.patch(
+    "general_superstaq.superstaq_client._SuperstaQClient.add_new_user",
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/user_config_test.py]: Done.

File: general_superstaq/user_config_test.py [hunk 6]
```diff
@@ -32,9 +32,52 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
...
+    return_value="The user has been added",
+)
+def test_add_new_user(  # pylint: disable=missing-function-docstring
+    mock_ibmq: mock.MagicMock,
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/user_config_test.py]: Done.

File: general_superstaq/user_config_test.py [hunk 7]
```diff
@@ -32,9 +32,52 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
...
+) -> None:
+    client = gss.superstaq_client._SuperstaQClient(
+        remote_host="http://example.com", api_key="key", client_name="general_superstaq"
+    )
+    service = gss.user_config.UserConfig(client)
+    assert service.add_new_user("Marie Curie", "mc@gmail.com") == "The user has been added"
+
+
+@mock.patch(
+    "general_superstaq.superstaq_client._SuperstaQClient.update_user_balance",
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/user_config_test.py]: Done.

File: general_superstaq/user_config_test.py [hunk 8]
```diff
@@ -32,9 +32,52 @@ def test_ibmq_set_token(  # pylint: disable=missing-function-docstring
...
+    return_value="The account's balance has been updated",
+)
+def test_update_user_balance(  # pylint: disable=missing-function-docstring
+    mock_ibmq: mock.MagicMock,
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/user_config_test.py]: Done.

File: general_superstaq/user_config.py [hunk 9]
```diff
@@ -22,14 +22,64 @@ def get_balance(self, pretty_output: bool = True) -> Union[str, float]:
...
+        return self._client.add_new_user(
+            {
+                "name": name,
+                "email": email,
+            }
+        )
+
+    def update_user_balance(self, email: str, balance: float) -> str:
+        """Updates user's balance.
+        Args:
```
Comment (richrines1) [inline general_superstaq/user_config.py:45]:
```suggestion

        Args:
```

File: general_superstaq/user_config.py [hunk 10]
```diff
@@ -22,14 +22,64 @@ def get_balance(self, pretty_output: bool = True) -> Union[str, float]:
...
+        return self._client.update_user_balance(
+            {
+                "email": email,
+                "balance": balance,
+            }
+        )
+
+    def update_user_role(self, email: str, role: int) -> str:
+        """Updates user's role.
+        Args:
```
Comment (richrines1) [inline general_superstaq/user_config.py:61]:
```suggestion

        Args:
```

**Review by richrines1 — APPROVED** [review]
two nits then lgtm!

File: general_superstaq/user_config.py (hunk 9 above)
Comment (vtomole) [reply general_superstaq/user_config.py:46]: Done.

File: general_superstaq/user_config.py (hunk 10 above)
Comment (vtomole) [reply general_superstaq/user_config.py:63]: Done.

**Pull request merged**